idx/memory/memory_test.go

```diff
@@ -925,6 +967,72 @@ func testSingleNodeMetric(t *testing.T) {
 	ix.AddOrUpdate(mkey, data, getPartition(data))
 }
 
+// TestMemoryIndexHeapUsageWithTags5 gives a rough estimate of how much memory the index is using with tag support turned on
+func TestMemoryIndexHeapUsageWithTags5(t *testing.T) {
+	testMemoryIndexHeapUsageWithTags(t, false, 5)
+}
+
+// TestMemoryIndexHeapUsageWithTagsAndUnique5 gives a rough estimate of how much memory the index is using with tag support turned on
+func TestMemoryIndexHeapUsageWithTagsAndUnique5(t *testing.T) {
+	testMemoryIndexHeapUsageWithTags(t, true, 5)
+}
+
+func testMemoryIndexHeapUsageWithTags(t *testing.T, uniqueTags bool, num int) {
+	// turn partitioning off
+	Partitioned = false
+
+	// turn tag support on
+	TagSupport = true
+
+	ix := New()
+	ix.Init()
+
+	tags := []string{
+		"mykey=myvalue",
+		"secondkey=anothervalue",
+		"thirdkey=onemorevalue",
+		"region=west",
+		"os=ubuntu",
+		"anothertag=somelongervalue",
+		"manymoreother=lotsoftagstointern",
+		"afewmoretags=forgoodmeasure",
+		"onetwothreefourfivesix=seveneightnineten",
+		"lotsandlotsoftags=morefunforeveryone",
+		"goodforpeoplewhojustusetags=forbasicallyeverything",
+	}
+	series := getMetricDataWithCustomTags(1, 2, num, 10, "somekindof.longereven.metricname.butinstead.ofashorterone.bunchofthingsandstuff", true, tags, uniqueTags)
+
+	t.Log("Initial memory stats after series has been allocated")
+	printMemUsage(t)
+
+	for _, data := range series {
+		mkey, err := schema.MKeyFromString(data.Id)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ix.AddOrUpdate(mkey, data, 0)
+	}
+
+	t.Log("Memory stats after AddOrUpdate called")
+	printMemUsage(t)
+
+	runtime.GC()
+	time.Sleep(3 * time.Second)
+
+	t.Log("Memory stats after runtime.GC() called")
+	printMemUsage(t)
+
+	// need this here, otherwise the index gets GCed
```

Comment:
you should be able to solve this by assigning it to a package global variable. can you try that?

Reply:
Assigning the index to a package level variable?

Reply:
if the concern is that the index will be GC'd , then assigning it to a variable that is declared at the package level (not within a function), should resolve that in a cleaner way then this